server/src/main/java/io/crate/execution/dml/ArrayIndexer.java

```diff
@@ -63,6 +75,16 @@ public void indexValue(List<T> values,
                     );
                 }
             }
+            if (tableVersionCreated.onOrAfter(Version.V_5_9_0)) {
+                // map '[]' to '_array_length_ = 0'
```

Comment:
Isn't this incompatible with <5.9?
AFAIK empty array has length NULL, see https://github.com/crate/crate/pull/12702

Reply:
I thought I would leave the indexing as generic as possible and handled it in `array_length` scalar implementation itself. Btw, `ArrayLengthQueryTest` should have one covering that, will double check.